central/cve/image/datastore/datastore_impl.go

```diff
@@ -116,71 +112,53 @@ func (ds *datastoreImpl) Suppress(ctx context.Context, start *types.Timestamp, d
 		return err
 	}
 
-	cves, missing, err := ds.storage.GetMany(ctx, ids)
+	vulns, err := ds.searcher.SearchRawCVEs(ctx, pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.CVE, cves...).ProtoQuery())
 	if err != nil {
 		return err
 	}
 
-	for _, cve := range cves {
-		cve.Suppressed = true
-		cve.SuppressActivation = start
-		cve.SuppressExpiry = expiry
+	for _, vuln := range vulns {
+		vuln.Suppressed = true
+		vuln.SuppressActivation = start
+		vuln.SuppressExpiry = expiry
 	}
-	if err := ds.storage.UpsertMany(ctx, cves); err != nil {
+	if err := ds.storage.UpsertMany(ctx, vulns); err != nil {
 		return err
 	}
 
-	ds.updateCache(cves...)
-
-	if len(missing) == 0 {
-		return nil
-	}
-	missingCVEs := make([]string, 0, len(ids))
-	for idx := range ids {
-		missingCVEs = append(missingCVEs, ids[idx])
-	}
-	return errors.Errorf("failed to snooze %d/%d cves. CVEs not found: %s", len(missing), len(ids), strings.Join(missingCVEs, ", "))
+	ds.updateCache(vulns...)
+	return nil
 }
 
-func (ds *datastoreImpl) Unsuppress(ctx context.Context, ids ...string) error {
-	// Image permission check replaced by vuln request permission check in 68.0. Previously, global image permission
-	// check did not guarantee that the requestor is disallowed from unsuppressing node/cluster cves. Hence, verification
-	// to determine if the cve is image cve was not added.
+func (ds *datastoreImpl) Unsuppress(ctx context.Context, cves ...string) error {
 	if ok, err := vulnRequesterOrApproverSAC.WriteAllowedToAll(ctx); err != nil {
 		return err
 	} else if !ok {
 		return sac.ErrResourceAccessDenied
 	}
 
-	cves, missing, err := ds.storage.GetMany(ctx, ids)
+	vulns, err := ds.searcher.SearchRawCVEs(ctx, pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.CVE, cves...).ProtoQuery())
```

Comment:
Can we just run a GetMany instead of a search?

Reply:
we have cves, not the ids. Therefore, we cannot.